main.go

```diff
@@ -767,6 +762,26 @@ func main() {
 					}
 				}
 				w.Flush()
+				f.Close()
+				if debug {
+					log.Println("updating data file", dataPath, "from tmp file")
+				}
+				err = os.Rename(dataTmp, dataPath)
+				if err != nil {
+					log.Println("Error renaming data file", err)
+					os.Remove(dataTmp)
+					os.Remove(dataPath)
+				}
+				// Now we can update the Prom data struct since the data has been saved,
+				// we don't want Prometheus to see a new target defined if the data's not ready.
+				// Update of new target json will happen within 10 seconds, that should be good enough
+				PromsLock.Lock()
+				_, promExists := Proms[prom.Hash]
+				Proms[prom.Hash] = *prom
+				PromsLock.Unlock()
+				if !promExists {
+					log.Println("New metric found", prom)
+				}
```

Comment:
Good idea in moving this to the bottom!  This way we can ensure it is fully flushed out before committing it!